ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -71,12 +71,14 @@ - (void)applyChangeset:(CKTransactionalComponentDataSourceChangeset *)changeset
 static void applyChangesToCollectionView(UICollectionView *collectionView,
                                          CKComponentDataSourceAttachController *attachController,
                                          NSMapTable<UICollectionViewCell *, CKTransactionalComponentDataSourceItem *> *cellToItemMap,
-                                         CKTransactionalComponentDataSourceState *currentState,
+                                         CKTransactionalComponentDataSourceState *state,
                                          CKTransactionalComponentDataSourceAppliedChanges *changes)
 {
-  [changes.updatedIndexPaths enumerateObjectsUsingBlock:^(NSIndexPath *indexPath, BOOL *stop) {
-    if (CKCollectionViewDataSourceCell *cell = (CKCollectionViewDataSourceCell *) [collectionView cellForItemAtIndexPath:indexPath]) {
-      attachToCell(cell, [currentState objectAtIndexPath:indexPath], attachController, cellToItemMap);
+  NSDictionary<NSIndexPath *, NSIndexPath *> *mapping = CKComputeFinalUpdatedIndexPathsForAppliedChanges(changes);
```

Comment:
Nit: `mapping` -> `updatedIndexPathsToFinalUpdatedIndexPaths`. A bit more verbose but it's a bit more explicit.